pandas/indexes/multi.py

```diff
@@ -368,15 +368,26 @@ def copy(self, names=None, dtype=None, levels=None, labels=None,
         ``deep``, but if ``deep`` is passed it will attempt to deepcopy.
         This could be potentially expensive on large MultiIndex objects.
         """
+        if names is not None and name is not None:
+            raise TypeError("Can only provide one of `names` and `name`")
+        if names is None:
+            names = name
```

Comment:
most/all of this can be pulled out of ``Base.copy`` and made into a private function, maybe ``_validate_name(name=name, names=names)`` (which you can then use in both ``.copy``
```
        names = kwargs.get('names')
        if names is not None and name is not None:
            raise TypeError("Can only provide one of `names` and `name`")
        if deep:
            from copy import deepcopy
            new_index = self._shallow_copy(self._data.copy())
            name = name or deepcopy(self.name)
        else:
            new_index = self._shallow_copy()
            name = self.name
        if name is not None:
            names = [name]
```